docs/guides/cdk/using-amplify-with-cdk.md

```diff
@@ -0,0 +1,275 @@
+      billingMode: dynamodb.BillingMode.PAY_PER_REQUEST,
+      partitionKey: {
+        name: 'id',
+        type: dynamodb.AttributeType.STRING,
+      },
+    });
+
+    /* Enables IAM access to the DynamoDB table from the Lambda function */
+    postTable.grantFullAccess(blogLambda)
+    
+    /* Creates an environment variable for consumption by the Lambda function */
+    blogLambda.addEnvironment('POST_TABLE', postTable.tableName);
+
+  }
+}
+```
+
+### Creating the GraphQL schema
+
+Create a __graphql__ folder in __cdk-backend__ and add a GraphQL schema at __cdk-backend/graphql/schema.graphql__
+
+```graphql
+type Post @aws_cognito_user_pools @aws_api_key {
+  id: ID!
+  title: String!
+  content: String!
+  username: String!
+}
+
+input PostInput {
+  id: ID!
+  title: String!
+  content: String!
+}
+
+type Mutation {
+  createPost(post: PostInput!): Post @aws_cognito_user_pools
+}
+
+type Query {
+  listPosts: [Post] @aws_cognito_user_pools @aws_api_key
+}
+```
+
+Next, create a folder called __lambda-fns__ in __cdk-backend__ and add a file called __main.ts__ within that folder with the following code:
+
+```ts
+const AWS = require('aws-sdk');
+const docClient = new AWS.DynamoDB.DocumentClient();
+
+type Post = {
+  id: string;
+  title: string;
+  content: string;
+  username: string;
+}
+
+type AppSyncEvent = {
+  info: {
+    fieldName: string
+  },
+  arguments: {
+    postId: string,
+    post: Post
+  },
+  identity: {
+    sub : string;
+    username : string;
+  }
+}
+
+async function createPost(post: Post) {
+    const params = {
+        TableName: process.env.POST_TABLE,
+        Item: post
+    }
+    try {
+        await docClient.put(params).promise();
+        return post;
+    } catch (err) {
+        console.log('DynamoDB error: ', err);
+        return null;
+    }
+}
+
+async function listPosts() {
+    const params = {
+        TableName: process.env.POST_TABLE,
+    }
+    try {
+        const data = await docClient.scan(params).promise()
+        return data.Items
+    } catch (err) {
+        console.log('DynamoDB error: ', err)
+        return null
+    }
+}
+
+exports.handler = async (event:AppSyncEvent) => {
+    const username = event.identity?.username;
+    switch (event.info.fieldName) {
+        case "createPost":
+            return await createPost({ ...event.arguments.post, username });
+        case "listPosts":
+            return await listPosts();
+        default:
+            return null;
+    }
+}
+```
+
+### Deploying the CDK back end & generating the exports file
+
+Now that the CDK back end is set up, you can now deploy it. When deploying a CDK back end, you can pipe any `CfnOutput` properties into a JSON file by using the `--outputs-file` flag.
+
+Next, deploy the infrastructure and pipe the CDK outputs file one directory up so it will be at the same level as the Amplify project:
+
+```sh
+npm run build && cdk deploy --outputs-file ../cdk-exports.json
```

Comment:
Error: `unable to determine cloud assembly asset output directory. Assets must be defined indirectly within a "Stage" or an "App" scope`

Reply:
ok this was caused by the CDK init adding different version than latest: 
![image](https://user-images.githubusercontent.com/6764957/98021804-b85fef80-1e3f-11eb-8c03-5c4f47ee9fa3.png)

Reply:
these are the deps i end up using

```
  "devDependencies": {
    "@aws-cdk/assert": "1.71.0",
    "@types/jest": "^26.0.10",
    "@types/node": "10.17.27",
    "jest": "^26.4.2",
    "ts-jest": "^26.2.0",
    "aws-cdk": "1.71.0",
    "ts-node": "^8.1.0",
    "typescript": "~3.9.7"
  },
  "dependencies": {
    "@aws-cdk/aws-appsync": "1.71.0",
    "@aws-cdk/aws-cognito": "1.71.0",
    "@aws-cdk/aws-dynamodb": "1.71.0",
    "@aws-cdk/aws-lambda": "1.71.0",
    "@aws-cdk/core": "1.71.0",
    "source-map-support": "^0.5.16"
  }
```